docs/development/vscode.md

```diff
@@ -0,0 +1,143 @@
+# Setting Up VS Code for Debugging Zulip
+
+## Overview
+
+Set up VS Code to debug the django zulip sever. The following can only
+setup VS Code to debug the *django application* and not the *JS*
+(frontend) part of zulip. This is still a work in progress and any
+suggestions to simplify or ease the workflow are encouraged.
+
+I hope this helps the community of VS Code users working on zulip and
+helps them significantly ! :D
+
+## Steps
+
+1. Download and install VS Code.
+2. Download and install the
+   [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+   extension for VS Code by Microsoft.  
+3. Set up the vagrant dev environment from [here](./setup-advanced.md)
+4. Make sure you run the dev server at least once. ( So that you know
+   everything is working before making changes to the files )
+5. Expose *port* in Vagrant file
+
+    Add the following line in `./Vagrantfile` and replace
+    `YOUR_GUEST_PORT_NUMBER` which you want to connect your VS Code
+    Debugger to the ptvsd server and `YOUR_HOST_PORT_NUMBER` to one on
+    which ptvsd server is going to run on.
+
+    **Note**: You can keep both same for simplicity
+    ```
+    config.vm.network "forwarded_port",
+    guest: YOUR_GUEST_PORT_NUMBER,
+    host: YOUR_HOST_PORT_NUMBER,
+    host_ip: host_ip_addr
+    ```
+    Example:
+    ```
+    config.vm.network "forwarded_port", guest: 5678, host: 5678, host_ip: host_ip_addr
+    ```
+
+6. Expose port in Docker
+
+    If you are using a *virtual box* instance, you can skip this step.
+    Add the following lines at the end of the
+    `tools/setup/dev-vagrant-docker` and replace
+    `YOUR_PTVSD_PORT_NUMBER` with the port number on which ptvsd server
+    will run. (This must be same as the `YOUR_HOST_PORT_NUMBER` from
+    last step)
+    ```docker
+    EXPOSE YOUR_PTVSD_PORT_NUMBER
+    ```
+    Example:
+
+    ```docker
+    EXPOSE 5678
+    ```
```

Comment:
Similarly, we should probably just make `tools/setup/dev-vagrant-docker` take care of this without any editing.

Reply:
How would we be doing that? I have made that change in `tools/setup/dev-vagrant-docker` itself. Do you mean we should add conditionals here?